### DISCUSSION TITLE
在同一个vue文件中，分别加载两个不同的vc-viewer,一个作为底图，一个供弹窗显示，多次打开弹窗显示地图后，会因为webgl泄露导致地图丢失

### DISCUSSION THREAD

**Discussion by xiaowunimte** [zouyaoji/vue-cesium#624] [Q&A]
![image](https://github.com/user-attachments/assets/2dcf4515-7121-451a-ae7d-563301274592)
![image](https://github.com/user-attachments/assets/f19aaf97-788f-403d-bfdd-90d87ff14aac)
![image](https://github.com/user-attachments/assets/a2cf3954-6be9-49a1-8569-cf6a674ae778)
外部v-if=false时，组件在卸载前也会让vc-viewer卸载，但还是避免不了这个问题

**Comment by xiaowunimte** [comment]
![image](https://github.com/user-attachments/assets/b1a8823c-3280-40b2-963a-69c2c901e73f)
补充

**Comment by xiaowunimte** [comment]
![image](https://github.com/user-attachments/assets/7b158931-a5dd-45a5-8b82-d2366437a701)
补充

**Comment by xiaowunimte** [comment]
![image](https://github.com/user-attachments/assets/8a2c4bd6-9638-4927-82f9-4551cea3bba6)
补充

**Comment by zouyaoji** [comment]
在vc-viewer组件上添加:remove-cesium-script="false" 即可避免类似问题。

试试

https://github.com/zouyaoji/vue-cesium/discussions/218

**Reply by xiaowunimte** [reply]
这个属性我已经设置过了，同时存在两个viewer实例时，其中一个频繁销毁创建就会出现这种情况，试过veiwer销毁也试过手动清除webgl上下文，好像解决不了这个问题

**Reply by xiaowunimte** [reply]
就是我想知道触发了destroy，是否完全销毁了viewer实例
![Uploading image.png…]()
